pennylane/transforms/qcut.py

```diff
@@ -939,6 +940,142 @@ def qcut_processing_fn_mc(
+    to be recombined to full bitstrings and if a classical processing function is supplied,
+    an expectation value will be evaluated.
+
+    Args:
+        tape (QuantumTape): the tape of the full circuit to be cut
+        shots (int): number of shots
+        device_wires (Wires):  wires of the device that the cut circuits are to be run on
+        max_depth (int): the maximum depth used to expand the circuit while searching for wire cuts
+        classical_processing_fn (callable): a classical postprocessing function to be applied to
+            the reconstructed bitstrings. The expected input is a bitstring; a flat array of length `wires`
+            and the output should be a single number within the interval [-1, 1]
+
+    Returns:
+        Tuple[Tuple[QuantumTape], Callable]: the tapes corresponding to the circuit fragments as a
+        result of cutting and a post-processing function which either recombines samples to a full
+        circuit bitstring or further calculates an expectation value if a classical processing function
+        is supplied
+
+    **Example**
+
+    The following :math:`3`-qubit circuit containins a :class:`~.WireCut` operation and a :func:`~sample`
+    measurement. When decorated with ``@qml.cut_circuit_mc``, we can cut the circuit into two
+    :math:`2`-qubit fragments:
+
+    .. code-block:: python
+
+        dev = qml.device("default.qubit", wires=2)
+
+        @qml.cut_circuit_mc(shots=shots, device_wires=[0, 1])
+        @qml.qnode(dev)
+        def circuit(x):
+            qml.RX(0.89, wires=0)
+            qml.RY(0.5, wires=1)
+            qml.RX(1.3, wires=2)
+
+            qml.CNOT(wires=[0, 1])
+            qml.WireCut(wires=1)
+            qml.CNOT(wires=[1, 2])
+
+            qml.RX(0.3, wires=0)
+            qml.RY(0.7, wires=1)
+            qml.RX(2.3, wires=2)
+            return qml.sample(wires=[0, 1, 2])
+
+    TODO: finish example with working code, add usage details
+
+    """
+
+    g = tape_to_graph(tape)
+    replace_wire_cut_nodes(g)
+    fragments, communication_graph = fragment_graph(g)
+    fragment_tapes = [graph_to_tape(f) for f in fragments]
+    fragment_tapes = [remap_tape_wires(t, device_wires) for t in fragment_tapes]
+
+    configurations, settings = expand_fragment_tapes_mc(
+        fragment_tapes, communication_graph, shots=shots
+    )
+
+    tapes = tuple(tape for c in configurations for tape in c)
+
+    if classical_processing_fn:
+        return tapes, partial(
+            qcut_processing_fn_mc,
+            communication_graph=communication_graph,
+            settings=settings,
+            shots=shots,
+            classical_processing_fn=classical_processing_fn,
+        )
+    else:
+        return tapes, partial(
+            qcut_processing_fn_sample, communication_graph=communication_graph, shots=shots
+        )
+
+
+@cut_circuit_mc.custom_qnode_wrapper
+def qnode_execution_wrapper(self, qnode, targs, tkwargs):
```

Comment:
I wonder if there is any issue with having two definitions for `qnode_execution_wrapper`? :thinking: I'm not against in principle, though if you see any weird issues then maybe this could be a cause?